### PR TITLE
Refactor user data directories with `platformdirs`

### LINKED ISSUE #100: Handle cache creation for multi-user envs differently
in 1.5.0alpha7, we have again an error where the cache DIR_PATH is used to create cache for next runs, a  multi-user install won't work because of permissions.

Example:

```python
---------------------------------------------------------------------------
PermissionError                           Traceback (most recent call last)
Cell In[10], line 5
      1 scenarios = [
      2     {"model": "image", "pathway":"SSP2-Base", "year": 2040},
      3 ]
----> 5 ndb = NewDatabase(
      6         scenarios = scenarios,        
      7         source_db="ecoinvent-3.8-cutoff",
      8         source_version="3.8",
      9         key='M2A0R2C2H322d470110934871348AFPDOQERPI77,
     10 )
     11 ndb.update_electricity()
     12 #ndb.write_db_to_brightway("test ME")

File /opt/tljh/user/envs/premise-bw2/lib/python3.10/site-packages/premise/ecoinvent_modification.py:492, in NewDatabase.__init__(self, scenarios, source_version, source_type, key, source_db, source_file_path, additional_inventories, system_model, system_args, use_cached_inventories, use_cached_database, external_scenarios, quiet, keep_uncertainty_data, gains_scenario)
    490 print("\n//////////////////// EXTRACTING SOURCE DATABASE ////////////////////")
    491 if use_cached_database:
--> 492     self.database = self.__find_cached_db(
    493         source_db, keep_uncertainty_data=keep_uncertainty_data
    494     )
    495     print("Done!")
    496 else:

File /opt/tljh/user/envs/premise-bw2/lib/python3.10/site-packages/premise/ecoinvent_modification.py:547, in NewDatabase.__find_cached_db(self, db_name, keep_uncertainty_data)
    540 """
    541 If `use_cached_db` = True, then we look for a cached database.
    542 If cannot be found, we create a cache for next time.
    543 :param db_name: database name
    544 :return: database
    545 """
    546 # check that directory exists, otherwise create it
--> 547 Path(DIR_CACHED_DB).mkdir(parents=True, exist_ok=True)
    548 # build file path
    549 if db_name is None:

File /opt/tljh/user/envs/premise-bw2/lib/python3.10/pathlib.py:1175, in Path.mkdir(self, mode, parents, exist_ok)
   1171 """
   1172 Create a new directory at this given path.
   1173 """
   1174 try:
-> 1175     self._accessor.mkdir(self, mode)
   1176 except FileNotFoundError:
   1177     if not parents or self.parent == self:

PermissionError: [Errno 13] Permission denied: '/opt/tljh/user/envs/premise-bw2/lib/python3.10/site-packages/premise/data/cache'

```

### PR DESCRIPTION
**Pull request by cmutel**
Fixes #100.

The current implementation stores user data in the `site-packages` directory. This is undesired, as:

* Users should not touch this directory in general
* It makes multi-user installations [problematic](https://github.com/polca/premise/issues/100) for many reasons

The alternative is to give each user a cache directory in a suitable location. We follow other Brightway libraries and user [platformdirs](https://platformdirs.readthedocs.io/en/latest/index.html), the successor to `appdirs`.

This PR also centralized the location of directory path creation, which should prevent inconsistencies, and provides one clear place where one can find filesystem constants: `filesystem_constants.py`.

### CODE REVIEW

**Comment by cmutel** [comment]
@romainsacchi @tngTUDOR I would love to get your input on the following:

The current export directory structure could also be refactored, so that the export directory wouldn't depend on the current working directory. Are you open to such a change? I have a solution in mind where it would be set by the user, by an environment variable, or would default to a platformdirs location.

This would also apply to the logs directory, which sits in the export hierarchy.

Is there a reason that `iam_variables_mapping` isn't part of the data directory?

**Comment by tngTUDOR** [comment]
~The changes are self-evident in code, but it would be worth it to add them to the CHANGELOG~

File: premise/__init__.py
Comment (tngTUDOR) [inline premise/__init__.py:1]: moves the DATA_DIR related vars to the new filesystem_constants module.

Comment (cmutel) [reply premise/__init__.py:1]: I changed all the internal imports, and these paths were in `__all__` so shouldn't be importable in any case?

File: premise/utils.py [hunk 1]
```diff
@@ -16,16 +16,12 @@
 from bw2io.importers.base_lci import LCIImporter
 from country_converter import CountryConverter
 from prettytable import ALL, PrettyTable
-from wurst.linking import (
-    change_db_name,
-    check_duplicate_codes,
-    check_internal_linking,
-    link_internal,
-)
+from wurst.linking import change_db_name, check_internal_linking, link_internal
```
Comment (tngTUDOR) [inline premise/utils.py:19]: which is the right black formatting ?

File: setup.py
Comment (tngTUDOR) [inline setup.py:1]: suggestion: let's keep only _1_ source of requirements (duplicated in `setup.py` and `requirements.txt`)

**Comment by tngTUDOR** [comment]
> @romainsacchi @tngTUDOR I would love to get your input on the following:
> 
> The current export directory structure could also be refactored, so that the export directory wouldn't depend on the current working directory. Are you open to such a change? I have a solution in mind where it would be set by the user, by an environment variable, or would default to a platformdirs location.
> 
> This would also apply to the logs directory, which sits in the export hierarchy.
> 
> Is there a reason that `iam_variables_mapping` isn't part of the data directory?

I agree that having a platformdirs folder, is an elegant way of centralizing files.

File: premise/utils.py (hunk 1 above)
Comment (cmutel) [reply premise/utils.py:19]: The CI does black auto-formatting, so we can't make this look any different in any case.

File: setup.py
Comment (cmutel) [reply setup.py:1]: Don't forget `meta.yaml` in `conda` :) Not sure that one can be anything but duplicative...

**Comment by cmutel** [comment]
The CI failing test is outside the changes I made (pandas variable name). @romainsacchi can I mark it skipped?

**Pull request merged**